feat(homepage): kenya map.

NofarGizra opened the pull request:
#34

Commit:
Basic

Commit:
Leaflet

Commit:
Map

Commit:
Map with Kenya

Commit:
Colors map (WIP)

NofarGizra commented:
Update:
Map is able to automatically focus on Kenya.
Now working on coloring areas (in progress).

Commit:
Now its working but it isn't written well...

Commit:
Separate CSS file

Commit:
fix app & ctrl names + css

Commit:
move 'ng-app' from html tag to body tag

Commit:
App is working, but this has no effect

Commit:
Best practices! (Map still not showing)

Commit:
Kenya with legend

Commit:
stuff...

Commit:
Blue map!

Commit:
COLORS!!! & openstreetmap

Commit:
remove debugs

Commit:
Kenya JSON & Remove helpers and debugs

Commit:
fix coords - now kenya is shown well!

Commit:
Change name of app (Merge branch '34')

NofarGizra commented:
Update:
Kenya is now displayed with all the different regions, but they are all the same color.

NofarGizra commented:
![iblimapp](https://cloud.githubusercontent.com/assets/3273122/3013680/7d718548-df4c-11e3-9a12-6dafb3f3c473.png)

NofarGizra commented:
Update:
We now have colors on the map, including a legend!
We also support the 2 different indexes - one for LRLD and one for SRSD. The will switch automatically according to the current date.

NofarGizra commented:
LRLD

![map1](https://cloud.githubusercontent.com/assets/3273122/3023361/90ff056c-dfe3-11e3-958d-ac92292c43c8.png)

NofarGizra commented:
SRSD

![map2](https://cloud.githubusercontent.com/assets/3273122/3023363/a052a5be-dfe3-11e3-9e9c-c73d33727c6a.png)

Commit:
Colors + Legend + 2 different seasons

niryariv commented:
awesome! 
![](http://pixel.nymag.com/imgs/daily/intelligencer/2014/02/09/olympics-gifs/putin-clap.o.jpg/a_560x375.jpg)

Commit:
README files

Commit:
DOCS

Commit:
Hovering a division now colors it white.

Commit:
(WIP) Display division name and %. Displaying, but not inside.

Commit:
Custom control displays name of division and percent on hover.

NofarGizra commented:
Update:
Added a custom block under the legend for displaying details.
When hovering a division, the division changes color to white, and its name and percents are displayed in the new custom block.

NofarGizra commented:
![map3](https://cloud.githubusercontent.com/assets/3273122/3037439/f957d84c-e0be-11e3-8f93-7b7700c76d2b.png)

amitaibu commented on apps/README.md:
```diff
@@ -0,0 +1,20 @@
+[![Build Status](https://magnum.travis-ci.com/Gizra/ethosia-client.svg?token=p2M1EeCrd3dY32WxWj3X&branch=master)](https://magnum.travis-ci.com/Gizra/ethosia-client)
+
+Ethosia Client
```
Remove this name here and other places

ceoaliongroo commented on apps/README.md:
```diff
@@ -0,0 +1,20 @@
+[![Build Status](https://magnum.travis-ci.com/Gizra/ethosia-client.svg?token=p2M1EeCrd3dY32WxWj3X&branch=master)](https://magnum.travis-ci.com/Gizra/ethosia-client)
```
Change to IBLI travis

ceoaliongroo commented on apps/app/css/main.css:
```diff
@@ -0,0 +1,56 @@
+/*
```
remove this file.

ceoaliongroo commented on apps/app/data-mock/geo.get.json:
```diff
@@ -0,0 +1,23 @@
+{
```
remove file.

ceoaliongroo commented on apps/app/index.html:
```diff
@@ -0,0 +1,83 @@
+    <!-- build:css({.tmp,app}) styles/main.css -->
+    <link rel="stylesheet" href="styles/main.css">
+    <!-- endbuild -->
+    <link rel="stylesheet" type="text/css" href="http://cloud.github.com/downloads/lafeber/world-flags-sprite/flags32.css" />
```
remove it.

ceoaliongroo commented on apps/app/index.html:
```diff
@@ -0,0 +1,83 @@
+        <script src="scripts/app.js"></script>
+        <script src="scripts/controllers/main.js"></script>
+        <script src="scripts/services/boxes.js"></script>
+        <script src="scripts/services/ethosiahttpinterceptor.js"></script>
```
- Change name of the file to: ibliHttpInterceptor.js
- Remove boxes.js

ceoaliongroo commented on apps/app/scripts/controllers/main.js:
```diff
@@ -0,0 +1,140 @@
+    });
+
+    // Get the CSV of indexes, according to the current season.
+    $http.get('csv/indexes' + currentSeason + '.csv').success(function(data) {
```
Move to a service.

ceoaliongroo commented on apps/app/scripts/services/ethosiahttpinterceptor.js:
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+angular.module('ibliApp')
+  .factory('ethosiaHttpInterceptor', function ($q, BACKEND_URL) {
```
replace ethosia => ibli

ceoaliongroo commented on apps/package.json:
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "ethosiaclient",
```
replace ethosia ==> ibli

Commit:
Code review fixes

Commit:
fix app name

Commit:
Create a service!!!! (WIP Stable but colors are wrong)

Commit:
Fix colors!!!!!

Commit:
docs & reorganize

Commit:
Add MaxBounds for Kenya.

NofarGizra commented:
Update:
- Technical stuff: Moved all data handling from the controller to a service.
- Added "maxBounds" - you now cannot have Kenya outside the map, it is always in the middle (even if you zoom in or zoom out).

ceoaliongroo commented on apps/app/scripts/controllers/main.js:
```diff
@@ -0,0 +1,51 @@
+    var hoverInfoControl = L.control();
+    hoverInfoControl.setPosition('bottomleft');
+    hoverInfoControl.onAdd = function () {
+      return $compile(angular.element('<hover-info></hover-info>'))($scope)[0];;
```
remove ;

ceoaliongroo commented on apps/app/scripts/controllers/main.js:
```diff
@@ -0,0 +1,51 @@
+    hoverInfoControl.setPosition('bottomleft');
+    hoverInfoControl.onAdd = function () {
+      return $compile(angular.element('<hover-info></hover-info>'))($scope)[0];;
+    }
```
add ;

ceoaliongroo commented on apps/app/styles/main.scss:
```diff
@@ -0,0 +1,128 @@
+    padding: 0.2em 0;
+}
+
+/* Space out content a bit */
```
Remove until line 97.

Commit:
Code review, docs...

ceoaliongroo commented on apps/app/scripts/directives/hover-info.js:
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+angular.module('ibliApp')
+  .directive('hoverInfo', function ($compile) {
```
Remove $compile, it's not needed.

Commit:
MOVE EVERYTHING INTO ibli

ceoaliongroo commented:
Merged.
